### ISSUE TITLE
Parse sqlite connection string

### ISSUE THREAD

**Issue by kirrg001** [vitaly-t/connection-string#1]
Hey 👋

Is that something you would like to support?
See https://github.com/TryGhost/knex-migrator/pull/74#issuecomment-335698861.

Thanks for your package!
Kate

**Comment by vitaly-t** [comment]
Are these the ones that are missing, or there's more to it? - 

> `sqlite://data.db` for a relative path.
`sqlite:///home/ghost/data.db` for an absolute path.

**Comment by kirrg001** [comment]
I think the tests [here](https://github.com/TryGhost/knex-migrator/pull/74/files#diff-fa125131ac8a2cfe4d9f6c0c1da334aeR7) reflect a good set of use cases, which should be supported. I think we don't have to support all of use cases we can think of - we can start with the regular once. What do you think?

**Comment by vitaly-t** [comment]
Cases that I can see there are not currently supported:

* `sqlite:file:/home/user/database.db`
* `sqlite:file:///C:/Documents%20and%20Settings/user/Desktop/database.db`
* `sqlite:/home/user/database.db`

All three contradict the current valid connection string syntax.

**Comment by vitaly-t** [comment]
b.t.w., one f the best specifications worth supporting: https://github.com/mongodb/specifications/blob/master/source/connection-string/connection-string-spec.rst

**Comment by vitaly-t** [comment]
Closing it, because I'm not planning to extend the library presently. Right now it is as generic as it can be, with standard clear syntax, and reliable parser.

**Comment by vitaly-t** [comment]
Version 0.6.0 received support for multiple hosts.

**Comment by vitaly-t** [comment]
And in order to keep the connection string as a valid URL Connection String, the file can be passed in as a parameter:

```js
const a = new ConnectionString('sqlite://?file=C%3A%2FDocuments%20and%20Settings%2Fuser%2FDesktop%2Fdatabase.db');

const file = a.params && a.params.file;
//=> C:/Documents and Settings/user/Desktop/database.db
```

And when needed to encode a file path into a valid URL value, you can either use the standard `encodeURIComponent` method:

```js
const s = encodeURIComponent('C:/Documents and Settings/user/Desktop/database.db');
//=> C%3A%2FDocuments%20and%20Settings%2Fuser%2FDesktop%2Fdatabase.db
```

or generate a connection string using the defaults:

```js
const a = new ConnectionString('sqlite://', {
    params: {
        file: 'C:/Documents and Settings/user/Desktop/database.db'
    }
});

const s = a.toString();
//=> sqlite://?file=C%3A%2FDocuments%20and%20Settings%2Fuser%2FDesktop%2Fdatabase.db
```

**Comment by kirrg001** [comment]
@vitaly-t Thanks 😍

**Comment by vitaly-t** [comment]
In addition, the latest version can understand syntax like this one:

```
sqlite:file:///home/user/database.db
```
as the only one on that list that does not break the standard URL Connection String syntax.

When used like this, you can do `protocol.split(':')` to get the list of protocols, and to get the path you can do `path.join('/')`.

**Comment by vitaly-t** [comment]
I have added the following question within the [FAQ] section:

* [How to use a file path in URL directly?](https://github.com/vitaly-t/connection-string/wiki/FAQ#how-to-use-a-file-path-in-url-directly)

[FAQ]:https://github.com/vitaly-t/connection-string/wiki/FAQ

**Comment by vitaly-t** [comment]
Another page added - [Compatibility](https://github.com/vitaly-t/connection-string/wiki/Compatibility).

**Comment by vitaly-t** [comment]
Released [v4.0.0](https://github.com/vitaly-t/connection-string/releases/tag/4.0.0), which now handles `+` correctly inside the protocol name.
